### PR TITLE
Added internal page redirect and redirect code dropdown

### PR DESCRIPTION
**Pull request by Praznik**
Added a SiteTree dropdown field so that you can redirect directly to internal pages, so that if the menu structure changes the redirect still works.

Added a redirect code dropdown so that you can select which redirect code you want to use, because 301 is volatile.
The default is still 301, so that it doesn't break anything.

### CODE REVIEW

**Comment by Praznik** [comment]
The weird thing is that no matter what I do I cannot seem to pass the Travis CI checks, all the errors are because it cannot connect to the db.

File: _config/redirectedurls.yml [hunk 1]
```diff
@@ -13,3 +13,5 @@ SilverStripe\CMS\Controllers\ModelAsController:
 SilverStripe\ORM\DatabaseAdmin:
   classname_value_remapping:
     RedirectedURL: 'SilverStripe\RedirectedURLs\Model\RedirectedURL'
+SilverStripe\RedirectedURLs\Model\RedirectedURL:
+  defaultRedirectCode: 301
```
Comment (wilr) [inline _config/redirectedurls.yml]: Could we follow the `lowercase_with_underscore` convention for this new option

**Comment by wilr** [comment]
Thanks for the contribution. If you could rebase, squash the commits so the history is clean I'll merge in and tag a `2.1.0`

File: _config/redirectedurls.yml (hunk 1 above)
Comment (Praznik) [reply _config/redirectedurls.yml]: Sure no problem.

**Comment by Praznik** [comment]
@wilr I squashed the commits and cherry picked teh 2 commits on to a new branch, I've made a new pull request since that is the only way I know how to do it 🤦‍♂

**Pull request closed without merging**